page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/fragments/FragmentDataTableUtils.java

```diff
@@ -0,0 +1,78 @@
+package ru.sbtqa.tag.pagefactory.fragments;
+
+import gherkin.ast.DataTable;
+import gherkin.ast.Node;
+import gherkin.ast.Step;
+import gherkin.ast.TableCell;
+import gherkin.ast.TableRow;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.reflect.FieldUtils;
+
+public class FragmentDataTableUtils {
+
+    private static final int HEADER_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+
+    private FragmentDataTableUtils() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static void applyDataTable(Step step, Step fragmentStep) throws IllegalAccessException {
+        Map<String, String> dataTableAsMap = getDataTableAsMap(step);
+
+        // apply datatable parameters to fragment step body
+        FieldUtils.writeField(fragmentStep, "text", applyToText(dataTableAsMap, fragmentStep.getText()), true);
+
+        // apply datatable parameters to fragment step argument data table
+        if (fragmentStep.getArgument() != null) {
+            FieldUtils.writeField(fragmentStep, "argument", applyToArgument(dataTableAsMap, fragmentStep), true);
+        }
+    }
+
+    private static Map<String, String> getDataTableAsMap(Step step) {
+        Map<String, String> dataTableAsMap = new HashMap<>();
+
+        Node argument = step.getArgument();
+        if (argument == null || !(argument instanceof DataTable)) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary null check; "instanceof" returns false for nulls. ([squid:S4201](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS4201))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=63&issues=AWSiSh4IMDLKieESKarT&open=AWSiSh4IMDLKieESKarT)

[ISSUE_KEY]: AWSiSh4IMDLKieESKarT